src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs

```diff
@@ -0,0 +1,69 @@
+﻿using MediatR;
+using SFA.DAS.PR.Application.Common.Commands;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Data;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+using SFA.DAS.ProviderRelationships.Types.Models;
+using System.Text.Json;
+
+namespace SFA.DAS.PR.Application.Permissions.Commands.RemovePermissions;
+
+public class RemovePermissionsCommandHandler(
+    IAccountProviderLegalEntitiesReadRepository _accountProviderLegalEntitiesReadRepository,
+    IPermissionsWriteRepository _permissionsWriteRepository,
+    IPermissionsAuditWriteRepository _permissionsAuditWriteRepository,
+    IProviderRelationshipsDataContext _providerRelationshipsDataContext
+) : IRequestHandler<RemovePermissionsCommand, ValidatedResponse<SuccessCommandResult>>
+{
+    public async Task<ValidatedResponse<SuccessCommandResult>> Handle(RemovePermissionsCommand command, CancellationToken cancellationToken)
+    {
+        AccountProviderLegalEntity? accountProviderLegalEntity = await _accountProviderLegalEntitiesReadRepository.GetAccountProviderLegalEntity(
+            command.Ukprn,
+            command.AccountLegalEntityId,
+            cancellationToken
+        );
+
+        if (accountProviderLegalEntity != null)
+        {
+            return await RemoveAndAuditPermissions(accountProviderLegalEntity, command, cancellationToken);
+        }
+
+        return new ValidatedResponse<SuccessCommandResult>();
+    }
+
+    private async Task<ValidatedResponse<SuccessCommandResult>> RemoveAndAuditPermissions(AccountProviderLegalEntity accountProviderLegalEntity, RemovePermissionsCommand command, CancellationToken cancellationToken)
+    {
+        var operationsToRemove = accountProviderLegalEntity.Permissions.Select(permission => permission.Operation).ToList();
```

Comment:
There is no need to convert this into a list, IEnumerable is serialisable I think

Reply:
fixed